src/components/MetadataPanel/index.tsx

```diff
@@ -0,0 +1,155 @@
+    selectedAgentMetadata,
+    uiDisplayData,
+}) => {
+    const [panelExpanded, setPanelExpanded] = React.useState(false);
+
+    const selectedAgentNotRendered =
+        selectedAgentMetadata.agentType >= uiDisplayData.length;
+
+    useEffect(() => {
+        if (!selectedAgentMetadata.uniqueId) {
+            setPanelExpanded(false);
+        } else {
+            setPanelExpanded(true);
+        }
+    }, [selectedAgentMetadata.uniqueId]);
+
+    const displayValues: Record<keyof AgentMetadata, MetadataDisplay> = {
+        uniqueId: {
+            label: <span>Unique ID</span>,
+            value: () => formatFloatForDisplay(selectedAgentMetadata.uniqueId),
+        },
+        agentType: {
+            label: <span>Agent Type</span>,
+            value: () => uiDisplayData[selectedAgentMetadata.agentType].title,
+        },
+        position: {
+            label: <span>Position</span>,
+            value: () => {
+                const pos = selectedAgentMetadata.position;
+                return (
+                    <>
+                        <div>x = {formatFloatForDisplay(pos.x)}</div>
+                        <div>y = {formatFloatForDisplay(pos.y)}</div>
+                        <div>z = {formatFloatForDisplay(pos.z)}</div>
+                    </>
+                );
+            },
+        },
+        rotation: {
+            label: (
+                <>
+                    Rotation
+                    <br />
+                    (degrees)
+                </>
+            ),
+            value: () => {
+                const rot = selectedAgentMetadata.rotation;
+                return (
+                    <>
+                        <div>x = {formatFloatForDisplay(rot.x)}</div>
+                        <div>y = {formatFloatForDisplay(rot.y)}</div>
+                        <div>z = {formatFloatForDisplay(rot.z)}</div>
+                    </>
+                );
+            },
+        },
+        radius: {
+            label: <span>Radius</span>,
+            value: () => formatFloatForDisplay(selectedAgentMetadata.radius),
+        },
+    };
+
+    const getMetadataRows = () => {
+        const keys = Object.keys(
+            selectedAgentMetadata
+        ) as (keyof AgentMetadata)[];
+        return keys.map((key, index) => (
+            <React.Fragment key={key}>
+                <div className={styles.row}>
+                    <div className={styles.key}>{displayValues[key].label}</div>
+                    <div>
+                        {selectedAgentNotRendered
+                            ? "-"
+                            : displayValues[key].value()}
+                    </div>
+                </div>
+                {index < keys.length - 1 && (
+                    <Divider className={styles.divider} />
+                )}
+            </React.Fragment>
+        ));
+    };
+
+    const panelContent = (): JSX.Element => {
+        if (!selectedAgentMetadata.uniqueId) {
```

Comment:
the other option here (although I think this is fine) is use `lodash.isEmpty(selectedAgentMetadata)`